interview-questions.md

```diff
@@ -8,36 +8,85 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-Your answer:
+Your answer: you'll fix it by doing a migration and adding the column of your foreign key in the Student model.
 
-Researched answer:
+Researched answer: If you created a model cohort that has many_students and forgot to add the foreign key you can use rails g migration AddCohort cohort:references  which will add a cohort_id to the students database column. Once you've created the migration you use the command rails db:migrate the students table will have a new a new foreign key that references the id column of the cohort table.
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: the use of params allows access to the attributes given in the model and by makes it easier to code in the RESTful routes.
 
-Researched answer:
+Researched answer: params is mostly seem pass in the RESTful routs update, show and delete.
 
 3. Name three rails generator commands. What is created by each?
 
-Your answer:
+Your answer: generate, db:, server
+- generate - will create the application 
+- db: - can be use to "create" a databese in the application, "migrate" to make adjustments in our database columns
+- server: will run a local server from our terminal to see a version of our app in a web brower by using the http://localhost:3000/
 
 Researched answer:
+rails controller = controller generators which are used to generate controller.
+rails g = model generator which we commonly use to create our models.
```

Comment:
Model and controllers are great examples! This specific question was asking for generator commands so things like migrations would work as well.